learn/quickstart/derived_types.md

```diff
@@ -0,0 +1,299 @@
+employ%last_name        = 'Doe'
+employ%city             = 'London' ! t_employ has access to t_address, because it inherits from t_person, that in return inherits from t_address
+employ%road_name        = 'BigBen'
+employ%house_number     = 1
+employ%position         = 'Intern'
+employ%monthly_salary   = 0.0
+end program test
+``` 
+
+## Options to declare members of a derived type
+
+`[member-variables]` refers to the declaration of all the member data types. These data types can be of any built-in data type, and/or of other derived types, as already show-cased in the above examples. However, member-variables can have their own extensive syntax, in form of:
+`type [,member-attributes] :: name[attr-dependent-spec][init]`
+
+`type`: any built-in type or other derived type
+
+`member-attributes` (optional):
+
+- `pointer` to specify a pointer
+- `allocatable` with or without `dimension` to specify a dynamic array
+- `public` or `private` access attributes
+- `protected` access attribute
+- `codimension` to specify a coarray 
+- `contiguous`
+
+{% include note.html content="`pointer` and `allocatable` cannot co-exist." %}
+
+{% include note.html content="`contiguous` requires an array with the `pointer` attribute." %}
+
+Examples for common cases:
+
+```fortran
+type :: t_example
+    !1st case: simple built-in type with access attribute and [init]
+    integer, private :: i = 0 ! private hides it from use outside of the t_example's scope. The default initialization [=0] is the [init] part. 
+
+    !2nd case: dynamic 1d_array
+    real, allocatable, dimension(:) :: x
+    ! the same as
+    real, allocatable :: x(:) ! parenthesis implies dimension(:) and is one of the possible [attr-dependent-spec].
+
+    !3rd case: protected
+    integer, protected :: i ! In contrary to private, protected allows access to i assigned value outside of t_example but is not definable, i.e. a value may be assigned to i only within t_example.
+
+    !4th case: pointer, with [init]
+    real, pointer :: x(:) => null() ! the [init] part is the [=>null()], pointers are discussed in the Advanced programming mini-book. 
+
+    !5th case: coarray
+    real, allocatable, codimension[:] :: z(:) ! a 1d_dynamic array shared in all threads. Coarrays will be discussed in the Advanced programming mini-book.
+    !or 
+    real, allocatable :: z(:)[:] ! here the [:] is the [attr-dependent-spec] and implies the codimension[:].
+
+    !6th case: contiguous
+    real, contiguous, pointer :: x(:)
+end type
+```
+
+{% include note.html content="In the above example the cases 4, 5 and 6 make use of Fortran `pointer` and `coarray` features that have not been addressed in this quickstart tutorial. However, they are presented here, in order for the readers to know that these features do exist and be able to recognize them. These features will be covered in detail in the upcoming `Advanced programing` mini-book." %}
```

Comment:
With Manning I learned a useful principle: "Don't introduce it until you have to use it". This only puts unnecessary burden on the reader. I suggest we move this section to the advanced tutorial.